File: CodingSamples/Java/Platform/AsyncParallelTest/Program1.java
import java.util.stream.*;

class Program {
	
	static class Computation {

		private long start;

		public long compute(int begin, int end) {
			start = System.currentTimeMillis();
			return IntStream.range(begin, end)
						.mapToLong(Worker::doWork)
						.sum();
		}

		public double time() {
			long stop = System.currentTimeMillis();
			return (stop - start) / 1000.0;
		}

	}

	public static void main(String[] args) throws Exception {
		int n = Integer.parseInt(args[0]);
		System.out.print("Computing...");
		var c = new Computation();
		long r = c.compute(1, n + 1);
		System.out.println("Done!");
		System.out.printf("Result = %d, computed in %.3f seconds.%n", r, c.time());
	}
}

File: CodingSamples/DotNet/Platform/AsyncParallelTest/DemoApp/Program1.cs
class Computation
{
    private System.Diagnostics.Stopwatch clock = new();

    public long Compute(int begin, int end)
    {
        clock.Start();
        return Enumerable.Range(begin, end - begin)
                .Select(Worker.DoWork)
                .Sum();
    }

    public double Time()
    {
        clock.Stop();
        return clock.Elapsed.TotalSeconds;
    }
}

class Program
{

    static void DoComputation(int limit)
    {
        Console.Write("Computing...");
        var c = new Computation();
        long r = c.Compute(1, limit + 1);
        Console.WriteLine("Done.");
        Console.WriteLine("Result = {0}, computed in {1:0.000} seconds.", r, c.Time());
    }

    static void Main(string[] args)
    {
        int n = int.Parse(args[0]);
        DoComputation(n);
    }
}
